### DISCUSSION TITLE
attu 400 Error: 14 UNAVAILABLE: No connection established

### DISCUSSION THREAD

**Discussion by wjswk** [milvus-io/milvus#24948] [Q&A and General discussion] [answered]
docker run --platform linux/amd64 -p 8000:3000  -e MILVUS_URL=127.0.0.1:19530 zilliz/attu:v2.2.5
yarn run v1.22.19
$ node dist/src/app.js
Attu server started: http://172.17.0.2:3000/api/v1/swagger/

 GET 304 /? 20.321 ms @ Fri, 16 Jun 2023 06:31:34 GMT ::ffff:172.17.0.1 from undefined Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36

 GET 304 /assets/index.78b3761c.js 6.660 ms @ Fri, 16 Jun 2023 06:31:34 GMT ::ffff:172.17.0.1 from undefined Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36

 GET 304 /assets/index.2034e75d.css 3.030 ms @ Fri, 16 Jun 2023 06:31:34 GMT ::ffff:172.17.0.1 from undefined Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36

 GET 200 /env-config.js 19.339 ms @ Fri, 16 Jun 2023 06:31:34 GMT ::ffff:172.17.0.1 from undefined Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36

 GET 304 /api/v1/milvus/version 14.090 ms @ Fri, 16 Jun 2023 06:31:34 GMT ::ffff:172.17.0.1 from undefined Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36

 GET 304 /attu.png 2.592 ms @ Fri, 16 Jun 2023 06:31:37 GMT ::ffff:172.17.0.1 from undefined Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36

Error: 14 UNAVAILABLE: No connection established
    at callErrorFromStatus (/app/node_modules/@grpc/grpc-js/build/src/call.js:31:19)
    at Object.onReceiveStatus (/app/node_modules/@grpc/grpc-js/build/src/client.js:192:76)
    at /app/node_modules/@grpc/grpc-js/build/src/call-interface.js:78:35
    at Object.onReceiveStatus (/app/node_modules/@zilliz/milvus2-sdk-node/dist/utils/Grpc.js:138:45)
    at Object.onReceiveStatus (/app/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:360:141)
    at Object.onReceiveStatus (/app/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:323:181)
    at /app/node_modules/@grpc/grpc-js/build/src/resolving-call.js:94:78
    at processTicksAndRejections (internal/process/task_queues.js:77:11)
for call at
    at ServiceClientImpl.makeUnaryRequest (/app/node_modules/@grpc/grpc-js/build/src/client.js:160:34)
    at ServiceClientImpl.<anonymous> (/app/node_modules/@grpc/grpc-js/build/src/make-client.js:105:19)
    at /app/node_modules/@zilliz/milvus2-sdk-node/dist/utils/Function.js:19:24
    at new Promise (<anonymous>)
    at promisify (/app/node_modules/@zilliz/milvus2-sdk-node/dist/utils/Function.js:16:15)
    at MilvusClient.<anonymous> (/app/node_modules/@zilliz/milvus2-sdk-node/dist/milvus/GrpcClient.js:227:72)
    at step (/app/node_modules/@zilliz/milvus2-sdk-node/dist/milvus/GrpcClient.js:59:23)
    at Object.next (/app/node_modules/@zilliz/milvus2-sdk-node/dist/milvus/GrpcClient.js:40:53)
    at /app/node_modules/@zilliz/milvus2-sdk-node/dist/milvus/GrpcClient.js:34:71
    at new Promise (<anonymous>) {
  code: 14,
  details: 'No connection established',
  metadata: Metadata { internalRepr: Map(0) {}, options: {} },
  expose: true,
  statusCode: 400,
  status: 400
}
POST /api/v1/milvus/connect 400 Error: 14 UNAVAILABLE: No connection established

 POST 400 /api/v1/milvus/connect 1278.131 ms @ Fri, 16 Jun 2023 06:31:42 GMT ::ffff:172.17.0.1 from undefined Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36

**Comment by yhmo** [comment, marked as answer]
Double-check the milvus server state. It might not be ready or failed to start.

**Reply by wjswk** [reply]
![image](https://github.com/milvus-io/milvus/assets/11325233/661f7989-3a10-4ba9-9b14-b84001d60474)

**Reply by yhmo** [reply]
How about using a pymilvus to test the connection?

**Reply by yhmo** [reply]
In the Attu README: https://github.com/zilliztech/attu

It mentioned that "Note that "127.0.0.1" or "localhost" will not work when running Attu on Docker." 
Use your local IP for the MILVUS_URL when you start the Attu, for example:
`docker run -p 8000:3000 -e MILVUS_URL=192.168.0.1:19530 zilliz/attu:v2.2.3`

**Reply by wjswk** [reply]
Because I opened VPN.
thanks

**Comment by soyisou** [comment]
![image](https://github.com/milvus-io/milvus/assets/116480575/02242281-40f5-4119-afc1-8f93e83e8387)

attu 400 Error: 14 UNAVAILABLE: No connection established

**Reply by yhmo** [reply]
The error message says "collection not loaded"
Milvus requires collection loaded into memory before any query.
Make the collection is loaded. If it is not loaded, click the "load" button to load it into memory.
![Unsaved Image 1](https://github.com/milvus-io/milvus/assets/2282099/9ddb3f16-837b-4e9f-8f55-acbb1fa9da10)
